OpenCL_C.txt

```diff
@@ -11687,25 +11687,25 @@ These conversions are performed as follows:
 
 `CL_UNORM_INT8` (8-bit unsigned integer) {rightarrow} `float`
 
-  :: normalized `float` value = `(float)c / 255.0f`
+  :: normalized `float` value = `round_to_float(c / 255)`
```

Comment:
Do we define round_to_float anywhere? I just realized there's no dedicated section in this spec corresponding to half floats, which makes me wonder if this section was intended to be generic to cover both float and half floats generically. If so, that could be clarified in the round_to_float definition, and it would be more clear to replace instances of `float` with something more generic like "floating-point" or "floating-point value".

Reply:
It looks like this is covered by https://github.com/KhronosGroup/OpenCL-Docs/pull/320/files. Only question then is a definition for round_to_float.

Reply:
I borrowed `round_to_float` from the SPIR-V Environment spec and the OpenCL C++ kernel language spec.  I've gone back and forth whether I like it or not, but I think it's OK if you think of the division being infinitely precise.  We should say this explicitly, though.

What if we did something like the conversions from float to int, and defined:

* The exact result, which is the division with no rounding.
* The rounded result, which is the exact result subject to some implementation-defined rounding.
* That the rounded result must be within 1.5 ULP of the exact result.

Would that work?

I like your suggestion to generalize this section so it applies to floats and halfs also.  I'll see if I can fold that into some other changes I've been experimenting with, to generalize the size of each normalized channel also.

Reply:
That sounds good. That leaves the CTS on the hook for computing the difference correctly, but I'm pretty sure computing the reference in double is sufficient. I don't remember how the CTS is handling this today.